### PR TITLE
Adding Linked List Implementation

### LINKED ISSUE #13: Implement a basic Linked List
- Create a new folder in DS implementations - name it Linked list and add its implementation. 
- Do corresponding additions in the documentation.
- Comment your code where necessary.

### PR DESCRIPTION
**Pull request by Vishal-raj-1**
# Description

I have implement a singly linked list in CPP. Fixed #13 .

## Type of change

Please check options that are relevant to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New Data Structure 
- [ ] New Algorithm

# Checklist:

- [x] I have squashed my commits
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My code is a working bug-free code

### CODE REVIEW

**Review by Manvityagi — CHANGES_REQUESTED** [review]
@Vishal-raj-1  Please do the necessary update in DOCUMENTATION.md also, See how other things are listed there.

File: DS implementations/Linked List/list.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+//typedef so we can simply use Node instead of struct Node
+typedef struct Node
+{
+    //To store the element
+    int data;
+    //To store the address of next element
+    struct Node *next;
+
+} Node;
```
Comment (ay2306) [inline DS implementations/Linked List/list.cpp:15]: Do not `typedef` for changing the name `Node` to `Node`.

**Review by ay2306 — CHANGES_REQUESTED** [review]
Please include all CRUD operation of Linked List

**Comment by Vishal-raj-1** [comment]
All conflicts and issues are resolved. @ay2306 @Manvityagi

**Comment by ay2306** [comment]
> All conflicts and issues are resolved. @ay2306 @Manvityagi

You have implemented the Create and Read node function only. Implement Update and Delete as well.

**Comment by Vishal-raj-1** [comment]
I have adding Delete and Reverse function.

**Review by ay2306 — CHANGES_REQUESTED** [review]
1. You haven't handled condition for `n` being greater than the size of the linked list for insert and delete operation
2. Implement Update for nth element as well.
3. You haven't squashed your commits, while you crossed the option in the PR description.

**Comment by Vishal-raj-1** [comment]
Sir i don't know how to squash commits, would you please help me a bit? I didn't get anything on google.

Thanks in advance.

**Comment by Vishal-raj-1** [comment]
Sir I have implement all functions. @ay2306

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Squash your commits @Vishal-raj-1

**Comment by Vishal-raj-1** [comment]
Squashing of commits is done succesfully. @Manvityagi @ay2306

**Comment by Manvityagi** [comment]
> Squashing of commits is done succesfully. @Manvityagi @ay2306
> 
> ss

The PR still shows 10 commits @Vishal-raj-1

**Comment by Vishal-raj-1** [comment]
I don't know what to do😢. In git bash all previous commits were sqaushed into first commit. Which is mark in screenshot.

sss

But it still show 10 commits on github.

ss2

**How to resolve this issue ?**

**Comment by Vishal-raj-1** [comment]
Could you please do squashing while merge this pull request ? @Manvityagi

**Comment by Vishal-raj-1** [comment]
Thanks for merging my pull request. @ay2306  @Manvityagi 😊

**Pull request merged**